src/main/java/gui/DashboardWindow.java

```diff
@@ -0,0 +1,307 @@
+                    jScrollPaneWithResults.setBounds(CENTER_CUSTOM_PHILOSOPHIES + 220, 600, 400, jScrollPanelHeight);
+                    add(jScrollPaneWithResults);
+                    System.out.println("eleoe");
+                }
+            }
+
+        }
+
+    }
+
+    public void updateCustomPhilosophiesList() {
+        List<String> customPhilosophiesNames = getCustomPhilosophiesNames();
+        int size = customPhilosophiesNames.size();
+        String[] strings = new String[size];
+        for (int i = 0; i < size; i++) {
+            strings[i] = customPhilosophiesNames.get(i);
+        }
+        ComboBoxModel<String> comboBoxModel = new DefaultComboBoxModel<>(strings);
+        jComboBoxCustomPhilosophies.setModel(comboBoxModel);
+    }
+
+    // poleci do innej klasy
+    private ImageIcon getStartingImageIcon() {
+        return getImageIcon(System.getProperty("user.dir") + "\\src\\main\\resources\\gui\\Blank_scenario.png");
+    }
+
+    // poleci do innej klasy
+    private ImageIcon getImageIcon(String path) {
+        BufferedImage img = null;
+        try {
+            img = ImageIO.read(new File(path));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Image dimg = img.getScaledInstance(IMAGE_WIDTH, IMAGE_HEIGHT,
+                Image.SCALE_SMOOTH);
+        return new ImageIcon(dimg);
+    }
+
+    //do innej klasy
+    public List<String> getCustomPhilosophiesNames() {
+        File f = new File(System.getProperty("user.dir") +
+                "\\src\\main\\resources\\gui\\customPhilosophies\\");
+        List<String> customPhilosophiesNames = new ArrayList<>();
+        for (String name : f.list()) {
+            customPhilosophiesNames.add(StringUtils.substringBefore(name, ".json"));
+        }
+        if (customPhilosophiesNames.size() == 0) {
+            customPhilosophiesNames.add("No custom philosophies yet");
+            isAnyCustomPhilosophy = false;
+        }
+        isAnyCustomPhilosophy = true;
+        return customPhilosophiesNames;
+    }
+
+    //do innej klasy
+    public CustomPhilosophy getCustomPhilosophyByName(String name) {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        File file = new File(System.getProperty("user.dir") +
+                "\\src\\main\\resources\\gui\\customPhilosophies\\" +
+                name);
+        CustomPhilosophy customPhilosophy = null;
+        try {
+            customPhilosophy = objectMapper.readValue(file, CustomPhilosophy.class);
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+        return customPhilosophy;
+    }
+
+    //potencjalnie do innej klasy, ale potrzeba też konsultacji jak będziemy to rozwiazywać
+    private String getActionNameFromDecision(String decisionString) {
+        String tmp = StringUtils.substringAfter(decisionString, "has_action: _");
+        return StringUtils.substringBefore(tmp, ";");
+    }
```

Comment:
Tutaj ogólnie dłuższy temat jak dostać ładną nazwę akcji. Na razie na froncie jest wyświetlana postać  w stylu change_lane_*_by_* | follow | turn_*. Mogę zrobić tak, żeby pozbyć się tych podkreśleń i było ładnie "Change lane right by 1"

Reply:
Ogólnie też testowałem jak są podpięte te akcje i wygląda to w taki sposób:
![image](https://user-images.githubusercontent.com/43828836/97857300-cc033d00-1cfd-11eb-9fb0-0f0ed3ebaf77.png)
Output:
```
<<<<
[follow()]
110_decision_2
<110_decision_2>
decision(has_action: _follow; )
8
><<<
<<<<
[turn_left()]
110_decision
<110_decision>
decision(has_action: _turn_left; )
7
><<<
<<<<
[action()]
110_decision_5
<110_decision_5>
decision(has_action: _change_lane_right_by_1; )
0
><<<
<<<<
[action()]
110_decision_3
<110_decision_3>
decision(has_action: _change_lane_left_by_1; )
0
><<<
<<<<
[action()]
110_decision_6
<110_decision_6>
decision(has_action: _change_lane_right_by_2; )
0
><<<
<<<<
[turn_right()]
110_decision_1
<110_decision_1>
decision(has_action: _turn_right; )
7
><<<
<<<<
[action()]
110_decision_4
<110_decision_4>
decision(has_action: _change_lane_left_by_2; )
0
><<<
```

Reply:
Także widać w sumie, że te akcje zmiany linii nie są jakoś inaczej podpinane niż to co było od dawna, czyli follow i turn left/right. Ogólnie to nic się nie dzieje, bo wyciągam nazwę z decision.toString(), które wygląda na przykład tak: "decision(has_action: _change_lane_left_by_2; )", ale warto wyłapać, że nie jest tak samo

Reply:
Nie da się gdzieś toString() nadpisać po prostu?

Reply:
problemem jest nadal jak toString() zaimplementować ;d Wydaje mi się, że jakikolwiek format, który będzie miał dobrą treść jest ok, możesz upiększyć usuwając '_' i nie męcz się z formatowanie stringów